### PR TITLE
Add Date/Time Metrics In Profiler Tab

### LINKED ISSUE #5435: Updating Profile Tab To Include New Metrics
For columns with fields (dates and timestamp), additional metric such as "Earliest" and "Latest" should be included as a metric to the Profiler tab.

### PR DESCRIPTION
**Pull request by nna077**
### Describe your changes :

These changes help find the latest time and earliest time in a Date/Time/Timestamp column. 
- Updated min/max to include handling for Date/Time/Timestamp fields
- Fixed the schema for min/max in `table.json` at catalog-rest-service directory to allow String and Integer types in Java
- Added unit tests to compute these fields

FIXES #5435
#
### Type of change :

- [] Bug fix
- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ingestion

### CODE REVIEW

File: catalog-rest-service/src/main/resources/json/schema/entity/data/table.json [hunk 1]
```diff
@@ -432,6 +432,14 @@
           "description": "Variance of a column.",
           "type": "number"
         },
+        "earliestTime": {
```
Comment (harshach) [inline catalog-rest-service/src/main/resources/json/schema/entity/data/table.json:435]: @nna077 Can you adde some details what are you trying to solve here?

Comment (nna077) [reply catalog-rest-service/src/main/resources/json/schema/entity/data/table.json:435]: Yes sure, These fields help find the latest time and earliest time in a Date/Time/Timestamp column. Say for example a service such as Trino has a table that contains three rows and a single column field which is of type: (Date). Row 1: 2020-01-01 Row 2: 2020-02-02 Row 3: 2020-03-03,  earliestTime = 2020-01-01 latestTime = 2020-03-03. This stat would be made available in the Profiler Tab. I have also attached a UI screenshot for clarification. 
image

**Comment by TeddyCr** [comment]
Hey @nna077 thank you for this PR. We were thinking adding support for datetime/date field to the min/max data quality tests would be the way to go. Would that be ok for you to rework the PR this way?

It would imply adding a is_datetime test [here](https://github.com/TeddyCr/OpenMetadata/blob/main/ingestion/src/metadata/orm_profiler/orm/registry.py) and then adding that logic in the min and max data quality tests [here](https://github.com/TeddyCr/OpenMetadata/blob/e9b6de9fa4b448bcc7b71fc15357e67dab451dea/ingestion/src/metadata/orm_profiler/metrics/static/min.py#L34-L38)

**Comment by sparekh-bbg** [comment]
> Hey @nna077 thank you for this PR. We were thinking adding support for datetime/date field to the min/max data quality tests would be the way to go. Would that be ok for you to rework the PR this way?
> 
> It would imply adding a is_datetime test [here](https://github.com/TeddyCr/OpenMetadata/blob/main/ingestion/src/metadata/orm_profiler/orm/registry.py) and then adding that logic in the min and max data quality tests [here](https://github.com/TeddyCr/OpenMetadata/blob/e9b6de9fa4b448bcc7b71fc15357e67dab451dea/ingestion/src/metadata/orm_profiler/metrics/static/min.py#L34-L38)

Hi @TeddyCr thanks for your comment. I work with @nna077 . Your suggestion seems to be a related but separate addition, right? The current PR is just to surface some info in the `Profiler` tab, similar to showing min/max for `int` and other numeric types. We are not so familiar with the quality checks to be honest, so if the project maintainers think both profiler and quality check need to go hand in hand, we can augment this PR with your suggestion.

**Comment by TeddyCr** [comment]
> Hi @TeddyCr thanks for your comment. I work with @nna077 . Your suggestion seems to be a related but separate addition, right? The current PR is just to surface some info in the `Profiler` tab, similar to showing min/max for `int` and other numeric types. We are not so familiar with the quality checks to be honest, so if the project maintainers think both profiler and quality check need to go hand in hand, we can augment this PR with your suggestion.

Hey @sparekh-bbg my bad I was referring to the profiler. By adding support for datetime metrics for the `min` and `max` this will surface this information in your column profile

**Comment by sparekh-bbg** [comment]
> > Hi @TeddyCr thanks for your comment. I work with @nna077 . Your suggestion seems to be a related but separate addition, right? The current PR is just to surface some info in the `Profiler` tab, similar to showing min/max for `int` and other numeric types. We are not so familiar with the quality checks to be honest, so if the project maintainers think both profiler and quality check need to go hand in hand, we can augment this PR with your suggestion.
> 
> Hey @sparekh-bbg my bad I was referring to the profiler. By adding support for datetime metrics for the `min` and `max` this will surface this information in your column profile

Thanks, will rework the PR to expose it via `min`/`max` instead.

**Comment by nna077** [comment]
> Hey @nna077 thank you for this PR. We were thinking adding support for datetime/date field to the min/max profiler would be the way to go. Would that be ok for you to rework the PR this way?
> 
> It would imply adding a is_datetime test [here](https://github.com/TeddyCr/OpenMetadata/blob/main/ingestion/src/metadata/orm_profiler/orm/registry.py) and then adding that logic in the min and max data quality tests [here](https://github.com/TeddyCr/OpenMetadata/blob/e9b6de9fa4b448bcc7b71fc15357e67dab451dea/ingestion/src/metadata/orm_profiler/metrics/static/min.py#L34-L38)

Hey I don't think this is feasible as the java classes generated from the jsonSchema are only of Double types, date/times will not be compatible

File: catalog-rest-service/src/main/resources/json/schema/entity/data/table.json [hunk 2]
```diff
@@ -402,11 +402,11 @@
         },
         "min": {
           "description": "Minimum value in a column.",
-          "type": ["number", "integer", "string"]
+          "oneOf": [{ "type": "number"}, { "type": "integer"}, { "type": "string"}]
```
Comment (pmbrull) [inline catalog-rest-service/src/main/resources/json/schema/entity/data/table.json:405]: Thanks for the fix, cleaner implementation. @harshach anything to add here?

**Review by pmbrull — APPROVED** [review]

**Pull request merged**